FastApi/app/routers/post.py

```
@@ -38,22 +40,39 @@ def get_posts(
     "/", status_code=status.HTTP_201_CREATED, response_model=schemas.PostResponse
 )
 def create_posts(
-    post: schemas.PostCreate,
+    file: UploadFile= File(None),
+    post: schemas.PostCreate = Depends(),
     db: Session = Depends(get_db),
     current_user: int = Depends(oauth2.get_current_user),
 ):
-
-    # # commiting changes into table
-    # conn.commit()
-
     print(current_user.id)
     print(current_user.email)
-    new_post = models.Post(owner_id=current_user.id, **post.model_dump())
-    db.add(new_post)
-    db.commit()
-    db.refresh(new_post)
-
-    return new_post
+    
+    url=None
+
+    try:
+        if file:
+            accepted_file_types = ["image/png", "image/jpeg", "image/jpg", "png", "jpeg", "jpg"]
+            file_type= file.content_type
+            print(file_type)
+
+            if file_type not in accepted_file_types:
+                return {"message": "Unsupported media!"}
```

You should raise an HTTPException here instead of returning the message